rust/connlib/tunnel/src/client.rs

```diff
@@ -77,66 +80,78 @@ where
     ) -> connlib_shared::Result<()> {
         self.role_state.add_resources(resources);
 
-        self.update_routes()?;
-        self.update_resource_list();
+        self.io
+            .device_mut()
+            .set_routes(self.role_state.routes().collect(), &self.callbacks)?;
+        self.callbacks
+            .on_update_resources(self.role_state.resources());
 
         Ok(())
     }
 
     pub fn remove_resources(&mut self, ids: &[ResourceId]) {
         self.role_state.remove_resources(ids);
 
-        if let Err(err) = self.update_routes() {
+        if let Err(err) = self
+            .io
+            .device_mut()
+            .set_routes(self.role_state.routes().collect(), &self.callbacks)
+        {
             tracing::error!(?ids, "Failed to update routes: {err:?}");
         }
 
-        self.update_resource_list();
-    }
-
-    fn update_resource_list(&self) {
         self.callbacks
-            .on_update_resources(self.role_state.resources());
+            .on_update_resources(self.role_state.resources())
     }
 
     /// Updates the system's dns
```

Comment:
```suggestion
    /// Updates our list of the system's default DNS resolvers
```
I thought this got merged in some other PR but now it's not there. We aren't setting the system DNS, we're setting connlib's knowledge of the system DNS, right?